crates/vertigo/src/computed/value.rs

```diff
@@ -27,49 +55,71 @@ impl<T: PartialEq + 'static> Clone for Value<T> {
 impl<T: PartialEq + 'static> Value<T> {
     pub fn new(deps: Dependencies, value: T) -> Value<T> {
         Value {
-            id: GraphId::default(),
-            value: Rc::new(BoxRefCell::new(Rc::new(value))),
-            deps
+            inner: Rc::new(BoxRefCell::new(
+                ValueInner {
+                    id: GraphId::default(),
+                    value: Rc::new(value),
+                    deps: deps.clone(),
+                },
+                "value inner"
+            )),
+            deps,
         }
     }
 
+    pub fn new_with_connect<F: Fn(&Value<T>) -> Box<dyn Any> + 'static>(deps: Dependencies, value: T, create: F) -> Computed<T> {
```

Comment:
This method can be moved to Dependencies impl, then `deps` parameter will be `self`.
It can be called `new_selfcomputed` or `new_selfcomputed_value`

Reply:
Done.